lib/connection/connection.js

```diff
@@ -38,175 +35,251 @@ var debugFields = [
- * @param {Buffer} [options.crl] SSL Certificate revocation store binary buffer
- * @param {Buffer} [options.cert] SSL Certificate binary buffer
- * @param {Buffer} [options.key] SSL Key file binary buffer
- * @param {string} [options.passphrase] SSL Certificate pass phrase
- * @param {boolean} [options.rejectUnauthorized=true] Reject unauthorized server certificates
- * @param {boolean} [options.promoteLongs=true] Convert Long values from the db into Numbers if they fit into 53 bits
- * @param {boolean} [options.promoteValues=true] Promotes BSON values to native types where possible, set to false to only receive wrapper types.
- * @param {boolean} [options.promoteBuffers=false] Promotes Binary BSON values to native Node Buffers.
+ * A class representing a single connection to a MongoDB server
+ *
  * @fires Connection#connect
  * @fires Connection#close
  * @fires Connection#error
  * @fires Connection#timeout
  * @fires Connection#parseError
- * @return {Connection} A cursor instance
+ * @fires Connection#message
  */
-var Connection = function(messageHandler, options) {
-  // Add event listener
-  EventEmitter.call(this);
-  // Set empty if no options passed
-  this.options = options || {};
-  // Identification information
-  this.id = _id++;
-  // Logger instance
-  this.logger = Logger('Connection', options);
-  // No bson parser passed in
-  if (!options.bson) throw new Error('must pass in valid bson parser');
-  // Get bson parser
-  this.bson = options.bson;
-  // Grouping tag used for debugging purposes
-  this.tag = options.tag;
-  // Message handler
-  this.messageHandler = messageHandler;
-
-  // Max BSON message size
-  this.maxBsonMessageSize = options.maxBsonMessageSize || 1024 * 1024 * 16 * 4;
-  // Debug information
-  if (this.logger.isDebug())
-    this.logger.debug(
-      f(
-        'creating connection %s with options [%s]',
-        this.id,
-        JSON.stringify(debugOptions(debugFields, options))
-      )
-    );
+class Connection extends EventEmitter {
+  /**
+   * Creates a new Connection instance
+   *
+   * @param {string} options.host The server host
+   * @param {number} options.port The server port
+   * @param {number} [options.family=null] IP version for DNS lookup, passed down to Node's [`dns.lookup()` function](https://nodejs.org/api/dns.html#dns_dns_lookup_hostname_options_callback). If set to `6`, will only look for ipv6 addresses.
+   * @param {boolean} [options.keepAlive=true] TCP Connection keep alive enabled
+   * @param {number} [options.keepAliveInitialDelay=300000] Initial delay before TCP keep alive enabled
+   * @param {boolean} [options.noDelay=true] TCP Connection no delay
+   * @param {number} [options.connectionTimeout=30000] TCP Connection timeout setting
+   * @param {number} [options.socketTimeout=360000] TCP Socket timeout setting
+   * @param {boolean} [options.singleBufferSerializtion=true] Serialize into single buffer, trade of peak memory for serialization speed
```

Comment:
Should be `singleBufferSerialization`.

Reply:
hah yeah, _should_ be removed! This option is no longer (actually has not been in some time) supported. Removed.